addOns/retire/src/main/java/org/zaproxy/addon/retire/model/Repo.java

```diff
@@ -214,15 +214,15 @@ private boolean dontcheck(HashMap<String, String> msginfo) {
         Extractors extractors = dc.getExtractors();
 
         // iterating over extractors
-        for (String criterion : msginfo.keySet()) {
-            matches = extractors.get(criterion);
+        for (Entry<String, String> criterion : msginfo.entrySet()) {
```

Comment:
Better leave with keySet, only keys are being used.

Reply:
This comment is wrong, the values are used, I missed that (the code was not changed to use the value). Could you restore and use the entry set to get the value?

Reply:
All good, got it